clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -617,118 +728,7 @@ SYCL::getDeviceLibraries(const Compilation &C, const llvm::Triple &TargetTriple,
-    if (SYCLTargetStr.starts_with("spir64_x86_64"))
-      return AOT_CPU;
-
-    if (SYCLTargetStr == "intel_gpu_pvc")
-      return AOT_PVC;
-
-    if (SYCLTargetStr.starts_with("intel_gpu_dg2"))
-      return AOT_DG2;
-
-    if (SYCLTargetStr.starts_with("spir64_gen")) {
-      ArgStringList TargArgs;
-      Args.AddAllArgValues(TargArgs, options::OPT_Xs, options::OPT_Xs_separate);
-      Args.AddAllArgValues(TargArgs, options::OPT_Xsycl_backend);
-      llvm::opt::Arg *A = nullptr;
-      if ((A = Args.getLastArg(options::OPT_Xsycl_backend_EQ)) &&
-          StringRef(A->getValue()).starts_with("spir64_gen"))
-        TargArgs.push_back(A->getValue(1));
-
-      return getSpecificGPUTarget(TargArgs);
-    }
-
-    return JIT;
-  };
-
-  std::string SanitizeVal;
-  size_t sanitizer_lib_idx = getSingleBuildTarget();
-  if (Arg *A = Args.getLastArg(options::OPT_fsanitize_EQ,
-                               options::OPT_fno_sanitize_EQ)) {
-    if (A->getOption().matches(options::OPT_fsanitize_EQ) &&
-        A->getValues().size() == 1) {
-      SanitizeVal = A->getValue();
-    }
-  } else {
-    // User can pass -fsanitize=address to device compiler via
-    // -Xsycl-target-frontend, sanitize device library must be
-    // linked with user's device image if so.
-    std::vector<std::string> EnabledDeviceSanitizers;
-
-    // NOTE: "-fsanitize=" applies to all device targets
-    auto SyclFEArgVals = Args.getAllArgValues(options::OPT_Xsycl_frontend);
-    auto SyclFEEQArgVals = Args.getAllArgValues(options::OPT_Xsycl_frontend_EQ);
-    auto ArchDeviceVals = Args.getAllArgValues(options::OPT_Xarch_device);
-
-    std::vector<std::string> ArgVals(
-        SyclFEArgVals.size() + SyclFEEQArgVals.size() + ArchDeviceVals.size());
-    ArgVals.insert(ArgVals.end(), SyclFEArgVals.begin(), SyclFEArgVals.end());
-    ArgVals.insert(ArgVals.end(), SyclFEEQArgVals.begin(),
-                   SyclFEEQArgVals.end());
-    ArgVals.insert(ArgVals.end(), ArchDeviceVals.begin(), ArchDeviceVals.end());
-
-    // Driver will report error if more than one of address sanitizer, memory
-    // sanitizer or thread sanitizer is enabled, so we only need to check first
-    // one here.
-    for (const std::string &Arg : ArgVals) {
-      if (Arg.find("-fsanitize=address") != std::string::npos) {
-        SanitizeVal = "address";
-        break;
-      }
-      if (Arg.find("-fsanitize=memory") != std::string::npos) {
-        SanitizeVal = "memory";
-        break;
-      }
-      if (Arg.find("-fsanitize=thread") != std::string::npos) {
-        SanitizeVal = "thread";
-        break;
-      }
-    }
-  }
-
-  if (SanitizeVal == "address")
-    addSingleLibrary(SYCLDeviceAsanLibs[sanitizer_lib_idx]);
-  else if (SanitizeVal == "memory")
-    addSingleLibrary(SYCLDeviceMsanLibs[sanitizer_lib_idx]);
-  else if (SanitizeVal == "thread")
-    addSingleLibrary(SYCLDeviceTsanLibs[sanitizer_lib_idx]);
+  addSYCLDeviceSanitizerLibs(C, IsSpirvAOT, LibSuffix, LibraryList);
```

Comment:
Could you add a comment as to why the device sanitizer libraries are not enabled for Windows builds?